assets/src/edit-story/components/panels/stylePreset/panel.js

```diff
@@ -0,0 +1,203 @@
+              ? fillColors
+              : fillColors.filter((color) => color !== toDelete),
+            textColors: !isText
+              ? textColors
+              : textColors.filter((color) => color !== toDelete),
+          },
+        },
+      });
+    },
+    [textStyles, fillColors, isText, textColors, updateStory]
+  );
+
+  const handleAddColorPreset = useCallback(
+    (evt) => {
+      evt.stopPropagation();
+      let addedPresets = {
+        fillColors: [],
+        textColors: [],
+        textStyles: [],
+      };
+      if (isText) {
+        addedPresets = {
+          ...addedPresets,
+          ...getTextPresets(selectedElements, stylePresets),
+        };
+      } else {
+        // Currently, shape only supports fillColors.
+        addedPresets = {
+          ...addedPresets,
+          ...getShapePresets(selectedElements, stylePresets),
+        };
+      }
+      if (
+        addedPresets.fillColors?.length > 0 ||
+        addedPresets.textColors?.length > 0 ||
+        addedPresets.textStyles?.length > 0
+      ) {
+        updateStory({
+          properties: {
+            stylePresets: {
+              textStyles: [...textStyles, ...addedPresets.textStyles],
+              fillColors: [...fillColors, ...addedPresets.fillColors],
+              textColors: [...textColors, ...addedPresets.textColors],
+            },
+          },
+        });
+      }
+    },
+    [
+      fillColors,
+      textStyles,
+      textColors,
+      isText,
+      selectedElements,
+      updateStory,
+      stylePresets,
+    ]
+  );
+
+  const handleApplyPreset = useCallback(
+    (preset) => {
+      if (isText) {
+        // @todo Determine this in a better way.
+        // Only style presets have background text mode set.
+        const isStylePreset = preset.backgroundTextMode !== undefined;
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: isStylePreset ? { ...preset } : { color: preset },
+        });
+      } else {
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: { backgroundColor: preset },
+        });
+      }
+    },
+    [isText, selectedElementIds, updateElementsById]
+  );
+
+  const colorPresets = isText ? textColors : fillColors;
+  const hasColorPresets = colorPresets.length > 0;
+  const hasPresets = hasColorPresets || textStyles.length > 0;
+
+  useEffect(() => {
+    // If there are no colors left, exit edit mode.
+    if (isEditMode && !hasPresets) {
+      setIsEditMode(false);
+    }
+  }, [hasPresets, isEditMode]);
+
+  // @todo This is temporary until the presets haven't been implemented fully with multi-selection.
+  if (!isText && !isShape && selectedElements.length > 1) {
+    return null;
+  }
+
+  const getEventHandlers = (preset) => {
+    return {
+      onClick() {
+        handlePresetClick(preset);
+      },
+      onKeyDown(evt) {
+        if (evt.keyCode === 'Enter' || evt.keyCode === 'Space') {
```

Comment:
Do you mean just `key`? The `keyCode` is always numeric, I believe. Also, `key` for space is `" "` (actual space char), I believe.

Reply:
Ah, right, thanks for noting. This (faulty) handling is actually generally not needed since the button takes care of it automatically.